### PR TITLE
Better verify cancellation tokens

### PR DESCRIPTION
**Pull request by LucGenetier**

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs [hunk 1]
```diff
@@ -286,8 +291,10 @@ protected static async Task<bool> MatchesAsync(RecordValue currentRecord, Record
                 return false;
             }
 
+            cancellationToken.ThrowIfCancellationRequested();
             await foreach (var baseRecordField in baseRecord.GetFieldsAsync(cancellationToken).ConfigureAwait(false))
             {
+                cancellationToken.ThrowIfCancellationRequested();
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs:297]: this one seems redundant with 295 (the one passed into the for loop)

Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs:297]: we're in an await foreach loop and not running in the same Task context so we need to check the CT again and in each loop iteration too, no?

File: src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs [hunk 2]
```diff
@@ -310,6 +317,7 @@ await foreach (var baseRecordField in baseRecord.GetFieldsAsync(cancellationToke
                 }
                 else if (baseRecordField.Value is RecordValue baseRecordValue && currentFieldValue is RecordValue currentRecordValue)
                 {
+                    cancellationToken.ThrowIfCancellationRequested();
```
[the remark this answers is not in the document]
Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs:320]: we are not in the same task context as line 297 due to line 298

[the remark this answers is not in the document]
Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs:320]: Looking at [https://devblogs.microsoft.com/premier-developer/recommended-patterns-for-cancellationtoken/](https://devblogs.microsoft.com/premier-developer/recommended-patterns-for-cancellationtoken/)

We are not passed the point of no cancellation here which is the next line, so I consider this check as legit here.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**